### PR TITLE
feat: Add support for none filter on 1:n relations.

### PR DESCRIPTION
**Pull request by SandPod**
### Adds
- Support for filtering on none for filtered or unfiltered many side in 1:n relations.

### Examples
Given the following entities:
``` company.yaml
class: Company
table: company
fields:
  name: String
  employees: List<Citizen>?, relation
```

``` citizen.yaml
class: Citizen
table: citizen
fields:
  name: String
```

> We can retrieve all companies that have no employees
``` dart
Company.find(
    session,
    where: (company) => company.employees.none()
);
```
> We can retrieve all companies that have no employees with a name starting with **a**
``` dart
Company.find(
    session,
    where: (company) => company.employees.none((e) => e.name.ilike("a%"))
);
```

> We can retrieve all companies with no employees with a name starting with **a** or no employees with a name starting with **b**
``` dart
Company.find(
    session,
    where: (company) => 
    company.employees.none((e) => e.name.ilike("a%")) 
    | company.employees.none((e) => e.name.ilike("b%"))
);
```

### Guidance for review
Best reviewed commit by commit.

Closes: https://github.com/serverpod/serverpod/issues/1404

## Pre-launch Checklist

- [x] I read the [Contribute](https://docs.serverpod.dev/contribute) page and followed the process outlined there for submitting PRs.
- [x] This update contains only one single feature or bug fix and nothing else. (If you are submitting multiple fixes, please make multiple PRs.)
- [x] I read and followed the [Dart Style Guide](https://dart.dev/guides/language/effective-dart/style) and formatted the code with [dart format](https://dart.dev/tools/dart-format).
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`), and made sure that the documentation follows the same style as other Serverpod documentation. I checked spelling and grammar.
- [x] I added new tests to check the change I am making.
- [x] All existing and new tests are passing.
- [x] Any breaking changes are documented below.

If you need help, consider asking for advice on the [discussion board](https://github.com/serverpod/serverpod/discussions).

## Breaking changes

None - this is a new expression method available on many relations.

### CODE REVIEW

File: packages/serverpod/lib/src/database/database_query.dart [hunk 1]
```diff
@@ -605,6 +664,7 @@ class _SubQuery {
 class _SubQueries {
   static const String orderByPrefix = 'order_by';
   static const String whereCountPrefix = 'where_count';
+  static const String whereNonePrefix = 'where_any';
```
Comment (SandPod) [inline packages/serverpod/lib/src/database/database_query.dart:667]: Chore: Wrong word used for prefix.
```suggestion
  static const String whereNonePrefix = 'where_none';
```

File: packages/serverpod/lib/src/database/database_query.dart [hunk 2]
```diff
@@ -683,26 +743,61 @@ class _SubQueries {
...
       /// Store index of the expressions so that we later can retrieve the
       /// matching sub query alias.
       expression.index = index;
+      if (expression is NoneExpression) {
+        subQueries[index] = _buildWhereNoneSubQuery(
+            relationQueryAlias, index, tableRelation, column, expression);
+      } else {
+        subQueries[index] = _buildWhereCountSubQuery(
+            relationQueryAlias, index, tableRelation, column, expression);
+      }
```
Comment (Isakdl) [inline packages/serverpod/lib/src/database/database_query.dart:756]: NIT: formatting add `,`

**Review by Isakdl — COMMENTED** [review]
Good stuff! Questions on a test but otherwise green!

**Review by Isakdl — APPROVED** [review]
LGTM!

**Pull request merged**